django/contrib/admin/utils.py

```diff
@@ -489,12 +489,32 @@ def construct_change_message(form, formsets, add):
     Translations are deactivated so that strings are stored untranslated.
     Translation happens later on LogEntry access.
     """
+    # `form.changed_data` can change depending on active language.
+    # Evaluating `form.changed_data` without disabling translations is
+    # necessary to avoid conflicts in the interpretation of `datetime`
+    # fields on the models.  For example, 12/3/2018 is 12th of March for
+    # a French person, but 3rd of December for a US person (default
+    # when translations are deactivated).
```

Comment:
Perhaps a less repetitive, less wordy version of this comment:
```python
    # Evaluating `form.changed_data` prior to disabling translations is required
    # to avoid fields affected by localization from being included incorrectly,
    # e.g. where date formats differ such as MM/DD/YYYY vs DD/MM/YYYY.
```